stock_barcodes/wizard/stock_barcodes_read_todo.py

```diff
@@ -208,3 +207,15 @@ def fill_from_pending_line(self):
             self.wiz_barcode_id[field] = self[field]
         self.wiz_barcode_id.product_uom_id = self.uom_id
         self.wiz_barcode_id._set_focus_on_qty_input()
+
+    def _get_package_product_dict(self, wiz_barcode, lines_list):
+        """Compute product into packages one time instead for every line
+        """
+        package_product_dic = {}
+        if wiz_barcode.option_group_id.source_pending_moves == "move_line_ids":
+            packages = self.env["stock.quant.package"].browse()
+            for lines in lines_list:
+                for line in lines:
+                    packages += line.package_id
```

Comment:
A more performant one should be:

```suggestion
                    package_ids = set()
                    for lines in lines_list:
                        for line in lines:
                            package_ids.add(line.package_id.id)
                    packages_product_dict = self.env["stock.quant.package"].browse(package_ids)._get_all_products_quantities()
                    
```

Reply:
@rousseldenis Changes done!! thanks